src/main/java/apoc/export/cypher/formatter/CypherFormatterUtils.java

```diff
@@ -0,0 +1,185 @@
+		Iterator<Label> labels = node.getLabels().iterator();
+		StringBuilder result = new StringBuilder(100);
+		boolean uniqueFound = false;
+		while (labels.hasNext()) {
+			Label next = labels.next();
+			String labelName = next.name();
+			if (uniqueConstraints.containsKey(labelName) && node.hasProperty(uniqueConstraints.get(labelName)))
+				uniqueFound = true;
+			if (indexNames.contains(labelName))
+				result.insert(0, label(labelName));
+			else
+				result.append(label(labelName));
+		}
+		if (!uniqueFound) {
+			result.append(label(UNIQUE_ID_LABEL));
+		}
+		return result.toString();
+	}
+
+    public static String notUniqueLabelsAsString(String id, Node node, Map<String, String> uniqueConstraints) {
+
+        String nodeLookupLabel = null;
+        for (Label l : node.getLabels()) {
+            String label = l.name();
+            String prop = uniqueConstraints.get(label);
+            if (prop == null) continue;
+            Object value = node.getProperty(prop, null);
+            if (value == null) continue;
+            nodeLookupLabel = label;
+        }
+
+        Iterator<Label> labels = node.getLabels().iterator();
+        StringBuilder result = new StringBuilder(100);
+        while (labels.hasNext()) {
+            Label next = labels.next();
+            String labelName = next.name();
+            if (labelName.equals(nodeLookupLabel)) continue;
+            result.append(", " + id + label(labelName));
+        }
+        return result.length() > 0 ? result.substring(2) : "";
+    }
+
+    // ---- properties ----
+
+	public static String propertiesAsString(String id, PropertyContainer pc, Set<String> indexedProperties, Long uniqueId, boolean jsonStyle) {
+		StringBuilder result = new StringBuilder(1000);
+		List<String> keys = Iterables.asList(pc.getPropertyKeys());
+		Collections.sort(keys);
+		for (String prop : keys) {
+			if (!indexedProperties.contains(prop)) continue;
+			result.append(formatPropertyName(id, prop, jsonStyle))
+					.append(toString(pc.getProperty(prop)));
+		}
+		for (String prop : keys) {
+			if (indexedProperties.contains(prop)) continue;
+			result.append(formatPropertyName(id, prop, jsonStyle))
+					.append(toString(pc.getProperty(prop)));
+		}
+		if (uniqueId != null) {
+			result.append(formatPropertyName(id, UNIQUE_ID_PROP, jsonStyle)).append(uniqueId);
+		}
+		return result.length() > 0 ? result.substring(2) : "";
+	}
+
+    public static String notUniquePropertiesAsString(String id, Node node, Map<String, String> uniqueConstraints, boolean jsonStyle) {
+
+        String nodeLookupProperty = null;
+        for (Label l : node.getLabels()) {
+            String label = l.name();
+            String prop = uniqueConstraints.get(label);
+            if (prop == null) continue;
+            Object value = node.getProperty(prop, null);
+            if (value == null) continue;
+            nodeLookupProperty = prop;
+        }
+
+        StringBuilder result = new StringBuilder(1000);
```

Comment:
this is also somehow copy & paste of a "formatProperties(node, Collection<String> skipProperties)" method that should be extracted